commit eed02e1bbf61
Author: Francois Marier <francois@mozilla.com>
Date:   Tue Jan 30 13:26:43 2018 -0800

    Bug 1431370 - Make DoRiceDeltaDecode allocation fallible to fix startup OOM crash. r=gcp a=RyanVM
    
    MozReview-Commit-ID: 2vf4EU4TVCq
    
    --HG--
    extra : source : bdb502c30c7c72c758d7bc4f22774fe27d045c1d
---
 toolkit/components/url-classifier/ProtocolParser.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git toolkit/components/url-classifier/ProtocolParser.cpp toolkit/components/url-classifier/ProtocolParser.cpp
index d4fce581e394..a5da314bda30 100644
--- toolkit/components/url-classifier/ProtocolParser.cpp
+++ toolkit/components/url-classifier/ProtocolParser.cpp
@@ -999,7 +999,10 @@ DoRiceDeltaDecode(const RiceDeltaEncoding& aEncoding,
 
   // Setup the output buffer. The "first value" is included in
   // the output buffer.
-  aDecoded.SetLength(aEncoding.num_entries() + 1);
+  if (!aDecoded.SetLength(aEncoding.num_entries() + 1, mozilla::fallible)) {
+    NS_WARNING("Not enough memory to decode the RiceDelta input.");
+    return NS_ERROR_OUT_OF_MEMORY;
+  }
 
   // Decode!
   bool rv = decoder.Decode(aEncoding.rice_parameter(),
